src/Microsoft.ML.DataView/TypeManager.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.ML.Internal.DataView;
+
+namespace Microsoft.ML.Data
+{
+    /// <summary>
+    /// A singleton class for managing the map between ML.NET <see cref="DataViewType"/> and C# <see cref="Type"/>.
+    /// To support custom column type in <see cref="IDataView"/>, the column's underlying type (e.g., a C# class's type)
+    /// should be registered with a class derived from <see cref="DataViewType"/>.
+    /// </summary>
+    public static class DataViewTypeManager
```

Comment:
Can you keep the file name and the class name sync'd? The file should be named `DataViewTypeManager.cs`.